reframe/utility/typecheck.py

```diff
@@ -129,15 +140,20 @@ def __getitem__(cls, elem_type):
             raise TypeError('invalid type specification for container type: '
                             'expected ContainerType[elem_type]')
 
-        ret = _ContainerType('%s[%s]' % (cls.__name__, elem_type.__name__),
-                             cls._bases, cls._namespace)
+        ret = _SequenceType('%s[%s]' % (cls.__name__, elem_type.__name__),
+                            cls._bases, cls._namespace)
         ret._elem_type = elem_type
-        ret.register_subtypes()
+        ret.register_container_type()
```

Comment:
Isn't this already called by the `__init__` method above?

Reply:
Yes, indeed. I removed it.